src/js/components/modals/AppModalComponent.jsx

```diff
@@ -38,7 +38,7 @@ var AppModalComponent = React.createClass({
     // is clicked, therefore it is held in the modal state rather than props
     return {
       app: app,
-      appIsValid: true,
+      appIsValid: false,
```

Comment:
The only difference is that I have to make changes before I can submit the edit form. Which is what the user should expect.